src/pkg/retention/policy/rule/latestk/evaluator.go

```diff
@@ -38,8 +38,12 @@ type evaluator struct {
 
 // Process the candidates based on the rule definition
 func (e *evaluator) Process(artifacts []*res.Candidate) ([]*res.Candidate, error) {
-	// TODO: REPLACE SAMPLE CODE WITH REAL IMPLEMENTATION
-	return artifacts, nil
+	i := e.k
+	if i > len(artifacts) {
+		i = len(artifacts)
+	}
+
+	return artifacts[:i], nil
```

Comment:
The input list does not guarantee the order. I think you need to do sort first and then return the k ones.